tests/functional/api/gateway_test.py

```diff
@@ -0,0 +1,89 @@
+from functools import partial
+
+import pytest
+from h_matchers import Any
+
+# pylint: disable=wildcard-import,unused-wildcard-import
+from tests.functional.oauth1 import *
+
+
+class TestGatewayHLTI:
+    def test_minimum_viable_login(self, gateway_launch, required_params):
+        response = gateway_launch(required_params)
+
+        assert response.headers["Content-Type"] == "application/json"
+        assert response.json == {
+            "h_api": {
+                "exchange_grant_token": {
+                    "data": {
+                        "assertion": Any.string(),
+                        "grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer",
+                    },
+                    "headers": {
+                        "Accept": "application/vnd.hypothesis.v2+json",
+                        "Content-Type": "application/x-www-form-urlencoded",
+                    },
+                    "method": "POST",
+                    "url": "https://example.com/api/token",
+                },
+                "list": {
+                    "headers": {"Accept": "application/vnd.hypothesis.v2+json"},
+                    "method": "GET",
+                    "url": "https://example.com/api/",
+                },
```

Comment:
Bit of duplication here, but it's easier than lots of Any.dict() stuff. Do you think we should cut this down?